### ISSUE TITLE
Simplex: update Wiki -> FAQ

### ISSUE THREAD

**Issue by peterrum** [dealii/dealii#11846]
Our [Wiki](https://github.com/dealii/dealii/wiki/Frequently-Asked-Questions#can-i-useimplement-trianglestetrahedra-in-dealii) states the following:

```
Can I use/implement triangles/tetrahedra in deal.II?

This is truly one of the most frequently asked questions. The short answer is: 
No, you can't. deal.II's basic data structures are too much tailored to 
quadrilaterals and hexahedra to make this trivially possible. Implementing 
other reference cells such as triangles and tetrahedra amounts to 
re-implementing nearly all grid and DoF classes from scratch, along with the 
finite element shape functions, mappings, quadratures and a whole host of other 
things. Making triangles and tetrahedra work would certainly involve having to 
write several ten thousand lines of code, and to make it usable in all the rest 
of the library would require auditing a very significant fraction of the 600,000 
lines of code that make up deal.II today.

That said, the current specialization on quadrilaterals and hexahedra has two 
very positive aspects: First, quadrilaterals and hexahedra typically provide a 
significantly better approximation quality than triangular meshes with the same 
number of degrees of freedom; you therefore get more accurate solutions for the 
same amount of work. Secondly, because the shape of cells are known, we can make 
a lot of things known to the compiler (such as the number of iterations of a 
loop over all vertices of a cell) which avoid a large number of run-time 
computations and makes the library as fast as it is. A simple example is that 
in deal.II we know that a loop over all vertices of a cell has exactly 
GeometryInfo<dim>::vertices_per_cell iterations, a number that is known to the 
compiler at compile-time. If we allowed both triangles and quadrilaterals, the 
loop would have cell->n_vertices() iterations, but this would in general not 
be known at compile time and consequently not allow the compiler to optimize on.

If you do need to work with a geometry for which all you have is a triangular 
or tetrahedral mesh, then you can convert this mesh into one that consists of 
quadrilaterals and hexahedra using the tethex program, see 
https://github.com/martemyev/tethex .
```

Before release 9.3, we need to update this entry. Maybe something like the following:

```
Can I use/implement triangles/tetrahedra in deal.II?

This is truly one of the most frequently asked questions. The short answer is: 
Yes, you can. Since release 9.3, we provide an experimental simplex and mixed
mesh support. To get started, see the tests (`tests/simplex`) and the 
[Simplex](https://www.dealii.org/developer/doxygen/deal.II/group__simplex.html) module page.

The implementation is quite mature (current limitations include that no adaptive-
mesh refinement is supported and only a few low-order finite elements are implemented)
but since it is quite new and not all functions in the library could be tested 
for the new mesh type (see also the old comment answering this same question about 
the complexity of making deal.II work on simplex meshes: "Making triangles and 
tetrahedra work [in deal.II] would certainly involve having to write several ten 
thousand lines of code, and to make it usable in all the rest of the library 
would require auditing a very significant fraction of the 600,000 lines of code 
that make up deal.II today."), we have labeled the simplex/mixed-mesh support as 
experimental. 
```

**Comment by marcfehling** [comment]
Looks good to me. Instead of referring to the tutorials (at the moment), you can mention the `tests/simplex` folder.

**Comment by peterrum** [comment]
This is the last remaining item from the `Release 9.3` milestone. Now that 9.3 is out and with that deal.II now also officially supports simplices, I think now is the right time to update the WIKI! Is the text fine with everyone?

**Comment by peterrum** [comment]
Done :)

**Comment by donggiang** [comment]
Hello #peterrum,
Following your comments above, I wonder How much work would I make to use mesh adaptivity with high order (p>=3) for triangular elements? I tried to add p=3 to the current implementation. It seems the code works well with nonadaptive mesh. So do you have any general advice if I want to work with mesh adaptivity for p>=3?
Second, could I renumbering dofs define on elements into a block for each  in the global system using a Discontinuos Galerkin Method? For my understanding, Deal.ii allows to renumber dofs of different variables into blocks, but it doesn't have the feature I mentione above.

**Comment by peterrum** [comment]
> I tried to add p=3 to the current implementation. 

That sounds great! Would you be willing to share the implementation and open a PR!? Have you also implemented the re-orientation tables? These are needed since we have multiple DoFs on the inside of faces in contrast to `p=1` (no DoFs) and `p=2` (one DoF).

> So do you have any general advice if I want to work with mesh adaptivity for p>=3?

There is already some support for AMR for simplex meshes:
1) in 2D, locally refined meshes should work for p=1/2 (see: https://github.com/dealii/dealii/blob/master/tests/multigrid-global-coarsening/multigrid_p_01.cc, https://github.com/dealii/dealii/blob/master/tests/simplex/hanging_nodes_01.cc, https://github.com/dealii/dealii/blob/master/tests/simplex/hanging_nodes_02.cc, https://github.com/dealii/dealii/blob/master/tests/simplex/hanging_nodes_03.cc, https://github.com/dealii/dealii/blob/master/tests/simplex/hanging_nodes_04.cc, https://github.com/dealii/dealii/blob/master/tests/simplex/hanging_nodes_hybrid_01.cc, https://github.com/dealii/dealii/blob/master/tests/simplex/hanging_nodes_hybrid_02.cc)
2) in 3D, you can locally refine meshes but the relevant constraints are still missing and need to be implemented (see: https://github.com/dealii/dealii/issues/11599)
3) in both cases, it is still not possible to coarsen cells 

If you have a working finite element for p=3, I wouldn't be surprised if 1) would work without changes or at most small changes. For 3), one needs to adjust `Triangulation` (and follow the exceptions that are thrown). Point 2) is a bit more involved. However, @drwells  and @elauksap are also interested in getting that work, so it might be a good idea if you talk to them so that work is not done redundantly!

**Comment by donggiang** [comment]
Thank you for your detailed explanations. 
I added calculations for p=3 in functions 'unit_support_points_fe_poly ( )' and 'get_fe_p_basis( )' in  'fe_simplex_p.cc' and 'polynomials_barycentric.cc' respectively (I may forget to mention others as I did 3 months ago). The latter I think is the reorientation tables you mentioned above.  I'm willing to share what I've done if you think those are useful. But what does 'opening a PR' mean ? :))
